test/powershell/Language/Parser/Conversions.Tests.ps1

```diff
@@ -20,4 +20,45 @@ Describe 'conversion syntax' -Tags "CI" {
         # This test relies on the fact that there are overloads (at least 2) for ToString method.
         ([System.Management.Automation.ActionPreference]"Stop").ToString() | Should Be "Stop"
     }
+
+    Context "Cast object[] to more narrow generic collection" {
+        BeforeAll {
+            $testCases1 = @(
+                @{ Command = {$result = [Collections.Generic.List[int]]@(1)};      CollectionType = 'List`1'; ElementType = "Int32";  Elements = @(1) }
+                @{ Command = {$result = [Collections.Generic.List[int]]@(1,2)};    CollectionType = 'List`1'; ElementType = "Int32";  Elements = @(1,2) }
+                @{ Command = {$result = [Collections.Generic.List[int]]"4"};       CollectionType = 'List`1'; ElementType = "Int32";  Elements = @(4) }
+                @{ Command = {$result = [Collections.Generic.List[string]]@(1)};   CollectionType = 'List`1'; ElementType = "String"; Elements = @("1") }
+                @{ Command = {$result = [Collections.Generic.List[string]]@(1,2)}; CollectionType = 'List`1'; ElementType = "String"; Elements = @("1","2") }
+                @{ Command = {$result = [Collections.Generic.List[string]]1};      CollectionType = 'List`1'; ElementType = "String"; Elements = @("1") }
+
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]@(1)};   CollectionType = 'Collection`1'; ElementType = "Int32"; Elements = @(1) }
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]@(1,2)}; CollectionType = 'Collection`1'; ElementType = "Int32"; Elements = @(1,2) }
+                @{ Command = {$result = [System.Collections.ObjectModel.Collection[int]]"4"};    CollectionType = 'Collection`1'; ElementType = "Int32"; Elements = @(4) }
+
+                @{ Command = {$result = [Collections.Generic.List[System.IO.FileInfo]]@('TestFile')};
+                    CollectionType = 'List`1'; ElementType = "FileInfo";  Elements = @('TestFile') }
+                @{ Command = {$result = [Collections.Generic.List[System.IO.FileInfo]]@('TestFile1', 'TestFile2')};
+                    CollectionType = 'List`1'; ElementType = "FileInfo";  Elements = @('TestFile1', 'TestFile2') }
+                @{ Command = {$result = [Collections.Generic.List[System.IO.FileInfo]]'TestFile'};
+                    CollectionType = 'List`1'; ElementType = "FileInfo";  Elements = @('TestFile') }
+            )
+        }
+
+        It "<Command>" -TestCases $testCases1 {
+            param($Command, $CollectionType, $ElementType, $Elements)
+
+            $result = $null
+            . $Command

```

Comment:
I see - neither is great - I'm fine either way then.
Because you tried what I suggested first - maybe it's worth a comment to save someone else the trouble trying to change it.